exposed-tests/src/test/kotlin/org/jetbrains/exposed/sql/tests/shared/ddl/EnumerationTests.kt

```diff
@@ -8,32 +8,40 @@ import org.jetbrains.exposed.sql.*
 import org.jetbrains.exposed.sql.tests.DatabaseTestsBase
 import org.jetbrains.exposed.sql.tests.TestDB
 import org.jetbrains.exposed.sql.tests.currentDialectTest
-import org.jetbrains.exposed.sql.tests.shared.DDLTests
 import org.jetbrains.exposed.sql.tests.shared.assertEquals
 import org.jetbrains.exposed.sql.vendors.H2Dialect
 import org.jetbrains.exposed.sql.vendors.MysqlDialect
 import org.jetbrains.exposed.sql.vendors.PostgreSQLDialect
 import org.junit.Test
+import org.postgresql.util.PGobject
 
 class EnumerationTests : DatabaseTestsBase() {
     private val supportsCustomEnumerationDB = TestDB.mySqlRelatedDB + listOf(TestDB.H2, TestDB.H2_PSQL, TestDB.POSTGRESQL, TestDB.POSTGRESQLNG)
 
+    internal enum class Foo {
+        Bar, Baz;
+
+        override fun toString(): String = "Foo Enum ToString: $name"
+    }
+
+    class PGEnum<T : Enum<T>>(enumTypeName: String, enumValue: T?) : PGobject() {
+        init {
+            value = enumValue?.name
+            type = enumTypeName
+        }
+    }
+
     object EnumTable : IntIdTable("EnumTable") {
-        internal var enumColumn: Column<DDLTests.Foo> = enumeration("enumColumn")
+        internal var enumColumn: Column<Foo> = enumeration("enumColumn")
 
         internal fun initEnumColumn(sql: String) {
             (columns as MutableList<Column<*>>).remove(enumColumn)
             enumColumn = customEnumeration(
                 "enumColumn", sql,
-                { value ->
-                    when {
-                        currentDialectTest is H2Dialect && value is Int -> DDLTests.Foo.entries[value]
-                        else -> DDLTests.Foo.valueOf(value as String)
-                    }
-                },
+                { value -> Foo.valueOf(value as String) },
```

Comment:
Based on this and the Wiki example being incorrect, maybe there was a reason for using Int when `customEnumeration` was first introduced but I couldn't write a test that fails without it.